crdt/causal_tree/operation_insert_counter.go

```diff
@@ -0,0 +1,85 @@
+package causal_tree
+
+import (
+	"encoding/json"
+	"fmt"
+	"strconv"
+
+	atm "github.com/crdteam/causal-tree/crdt/atom"
+)
+
+// +------------------------------+
+// | Operations - Insert Add atom |
+// +------------------------------+
+
+//Inserts an add atom as a child of the atom pointed by cursor.
+type InsertAdd struct {
+	//Value inserted into the counter container
+	Value int32
+}
+
+func (v InsertAdd) AtomPriority() int { return insertAddPriority }
+func (v InsertAdd) MarshalJSON() ([]byte, error) {
+	return json.Marshal(fmt.Sprintf("insert %d", v.Value))
+}
+
+func (v InsertAdd) String() string { return strconv.FormatInt(int64(v.Value), 10) }
+
+//InsertAdd atoms only accept child of type InsertAdd.
+func (v InsertAdd) ValidateChild(child atm.AtomValue) error {
+	switch child.(type) {
+	case InsertAdd:
+		return nil
+	default:
+		return fmt.Errorf("invalid atom value after InsertAdd: %T (%v)", child, child)
+	}
+}
+
+// InsertAdd inserts an InsertAdd atom after the cursor position and advances the cursor.
+func (t *CausalTree) InsertAdd(val int32) error {
+	atomID, err := t.addAtom(InsertAdd{val})
+	if err != nil {
+		return err
+	}
+	t.Cursor = atomID
+	return nil
+}
+
+// InsertAddAt inserts an InsertAdd atom after the given (tree) position.
+func (t *CausalTree) InsertAddAt(val int32, i int) error {
+	if err := t.SetCursor(i); err != nil {
+		return err
+	}
+	return t.InsertAdd(val)
+}
+
+// +---------------------------------------+
+// | Operations - Insert counter container |
+// +---------------------------------------+
+
+//Inserts a counter container as a child of the root atom.
```

Comment:
Please run `go fmt` to fix this and other comments automatically.

Reply:
done